### DISCUSSION TITLE
dose pyinstaller pack python module that not used?

### DISCUSSION THREAD

**Discussion by RonaldinhoL** [pyinstaller/pyinstaller#6123] [Help] [answered]
i project include a lib consistant of "server" and "client" part, i use "client" part and want to pack it to a single executable file, i wonder if it will pack "server" part? i do't want it to be leak or i will delete it manually from source code.

**Comment by rokm** [comment, marked as answer]
That depends on what "not used" means. If the server part of the package is imported in any of the modules that your code ends up importing (or if those modules end up importing the server-related modules), the server part will get packaged as well (for example, if the package's top-level `__init__.py` imports both the server and the client side, both will be packaged).

You can always check `build/<name>/Analysis-00.toc` (or `PYZ-00.toc`) to see what python modules ended up collected. You could also explicitly exclude the server-related modules using [`--exclude-module`](https://pyinstaller.readthedocs.io/en/latest/usage.html?highlight=--exclude-module#cmdoption-exclude-module), but that will work only if they are conditionally imported in the modules that triggered their inclusion in the first place (otherwise those will raise a missing-module error).

**Reply by RonaldinhoL** [reply]
great! thank you very much :)
